### PR TITLE
Support rrweb mask all inputs

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

Submitting this super quickly before a call just so I don't forget.

We might want to consider supporting other rrweb config props as well.

User request.

## Checklist
- [ ] Tests for new code (if applicable)
- [ ] TypeScript definitions (module.d.ts) updated and in sync with library exports (if applicable)

### CODE REVIEW

File: src/posthog-core.js [hunk 1]
```diff
@@ -83,6 +83,7 @@ const defaultConfig = () => ({
     inapp_link_new_window: false,
     request_batching: true,
     properties_string_max_length: 65535,
+    mask_all_inputs: false,
```
Comment (macobo) [inline src/posthog-core.js:86]: As-is this option is confusing. We should clearly indicate this is session recording specific.

How about:

```
session_recording?: {
  mask_all_inputs: false
},
```

File: src/module.d.ts [hunk 2]
```diff
@@ -523,6 +523,7 @@ declare namespace posthog {
         request_batching?: boolean
         sanitize_properties?: (properties: posthog.Properties, event_name: string) => posthog.Properties
         properties_string_max_length?: number
+        mask_all_inputs?: boolean
```
Comment (macobo) [inline src/module.d.ts:526]: Please keep the docstrings in sync - there's one in module.d.ts and one in posthog-core explaining all of the options.

**Comment by yakkomajuri** [comment]
Thanks for the review! Thought I'd get the comment about the docstrings, I just put out fast PRs because my no-no-meeting days are so hectic. 

Also thought of allowing users to pass all props but I guess we need to regulate the canvas one at least.

**Comment by yakkomajuri** [comment]
@macobo I'll put out a separate PR to handle all rrweb options. `mask_all_inputs` turned out to be useful beyond session recording too as per https://github.com/PostHog/posthog/issues/3770

File: src/autocapture.js [hunk 3]
```diff
@@ -43,7 +43,9 @@ var autocapture = {
         _.each(elem.attributes, function (attr) {
             // Only capture attributes we know are safe
             if (isSensitiveElement(elem) && ['name', 'id', 'class'].indexOf(attr.name) === -1) return
-            if (shouldCaptureValue(attr.value)) {
+
+            console.log('!!!!!!!', maskInputs)
```
Comment (macobo) [inline src/autocapture.js]: Debug logging

File: src/__tests__/autocapture.js [hunk 4]
```diff
@@ -727,14 +735,57 @@ describe('Autocapture system', () => {
...
+            }
+            autocapture._captureEvent(e1, lib)
+            const props1 = getCapturedProps(lib.capture)
+            expect(props1['$elements'][0]).toHaveProperty('$el_text')
+            expect(props1['$elements'][0]['$el_text']).toMatch('')
+        })
     })
 
     describe('_addDomEventHandlers', () => {
+        const sandbox = sinon.createSandbox()
```
Comment (macobo) [inline src/__tests__/autocapture.js:776]: Note - you're using old-style (non-jest) mocking - this was probably way harder than it needed to be. Doesn't block this PR though

**Review by macobo — APPROVED** [review]
Some code notes, but lgtm otherwise. Don't forget to bump the version!

File: src/__tests__/autocapture.js (hunk 4 above)
Comment (yakkomajuri) [reply src/__tests__/autocapture.js:776]: oh right - I just leveraged existing tests as a template

**Comment by lharress** [comment]
Sorry to jump in, following on from https://github.com/PostHog/posthog/issues/3770. I took a closer look at this to see if it solves another problem for us (one we intended to solve with `ph-no-capture`).

If I'm reading this correctly, it prevents collection of all attributes if `true`? Not just `input` elements?

**Comment by lharress** [comment]
Thanks! Helps us with another security issue :-)

**Pull request merged**